### PR TITLE
AK: Update HashTables head and tail when shifting during deletion

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
+ 2 smaller changes to make the code a bit nicer

### CODE REVIEW

**Comment by GMTA** [comment]
@Hendiadyoin1 nice find! Please add a test to expose this issue though :^)

File: AK/HashTable.h [hunk 1]
```diff
@@ -409,8 +409,17 @@ class HashTable {
...
     {
         VERIFY(!is_empty());
-        T element = *m_collection_data.tail->slot();
-        remove(element);
+        T element = move(*m_collection_data.tail->slot());
+        delete_bucket(*m_collection_data.tail);
+        return element;
+    }
+
+    T take_first()
```
Comment (GMTA) [inline AK/HashTable.h:417]: To be consistent, I think we should have `pop` / `shift` _or_ `take_first` / `take_last`

**Review by GMTA — CHANGES_REQUESTED** [review]

File: AK/HashTable.h (hunk 1 above)
Comment (Hendiadyoin1) [reply AK/HashTable.h:417]: I was wondering what the counterpart to `pop` was called...

Vector uses `take`, and I think that is more intuitive, so if you don't mind I will change it to that

Comment (GMTA) [reply AK/HashTable.h:417]: > so if you don't mind

It wasn't my API design to begin with so knock yourself out!

File: Tests/AK/TestHashTable.cpp [hunk 2]
```diff
@@ -5,6 +5,7 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include "AK/Forward.h"
```
Comment (GMTA) [inline Tests/AK/TestHashTable.cpp:8]: Please move this down as `#include <AK/Forward.h>` to match the others

Comment (GMTA) [inline Tests/AK/TestHashTable.cpp:8]: Do we need this include? And if so, can you change it to `#include <AK/Forward.h>` to match the others?

Comment (Hendiadyoin1) [reply Tests/AK/TestHashTable.cpp:8]: Ah I hate my clangd, adding includes without being asked to do so

**Review by GMTA — APPROVED** [review]

**Comment by Hendiadyoin1** [comment]
Ah missed at least one pop,
will push shortly, after my local build finishes

**Pull request merged**